src/js/jsx/shared/ColorPicker.jsx

```diff
@@ -403,18 +406,121 @@ define(function (require, exports, module) {
                     onMouseUp={this._handleMouseUp}
                     onMouseDown={this._handleMouseDown}
                     onTouchStart={this._startUpdates}>
+                    <div
+                        className="color-picker-map__background"
+                        style={{
+                            backgroundColor: this.props.backgroundColor
+                        }} />
+                    <div
+                        className="color-picker-map__pointer"
+                        style={{
+                            left: this._getPercentageValue(this.props.x),
+                            bottom: this._getPercentageValue(this.props.y)
+                        }} />
+                </div>
+            );
+        }
+    });
+    /**
+    * Placeholder for color type, no-color and color swatch.
+    * 
+    * @constructor
+    */
+    var ColorType = React.createClass({
+        render: function () {
+            return (
                 <div
-                    className="color-picker-map__background"
-                    style={{
-                        backgroundColor: this.props.backgroundColor
-                    }} />
-                <div
-                    className="color-picker-map__pointer"
-                    style={{
-                        left: this._getPercentageValue(this.props.x),
-                        bottom: this._getPercentageValue(this.props.y)
-                    }} />
-              </div>
+                    className="color-picker__colortype">
+                    <div
+                        className="color-picker__colortype__thumb empty">
+                    </div>
+                    <Gutter/>
+                    <div
+                        className="color-picker__colortype__thumb selected">
+                    </div>
+                    <Gutter/>
+                    <Gutter/>
+                    <NumberInput
+                            ref="left"
+                            size="column-11"
+                            placeholder="rgba(255,255,255,1)"
+                            value="rgba(255,255,255,1)"/>
+                  </div>
+            );
+        }
+    });
+    /**
+     * Placeholder color RGB and HSB input values.
+     * 
+     * @constructor
+     */
+    var ColorFields = React.createClass({
+        render: function () {
+            return (
+                <div className="color-picker__rgbhsb">
+                    <div className="formline">
+                        <Label>
+                            R
+                        </Label>
+                        <Gutter/>
+                        <NumberInput
+                            ref="left"
```

Comment:
Unclear, we might need them to implement the color input: https://facebook.github.io/react/docs/more-about-refs.html

The refs just need to be unique